bedrock/src/backup/client_events.rs

```diff
@@ -0,0 +1,421 @@
+use bedrock_macros::{bedrock_error, bedrock_export};
+use rand::RngCore;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use strum::Display;
+
+use super::manifest::ManifestManager;
+use crate::backup::manifest::BackupManifest;
+use crate::backup::BackupFileDesignator;
+use crate::primitives::config::Os;
+use crate::primitives::filesystem::{
+    create_middleware, get_filesystem_raw, FileSystemExt, FileSystemMiddleware,
+};
+use crate::primitives::http_client::{get_http_client, HttpHeader};
+use crate::HttpMethod;
+
+/// Errors that can occur when reporting client events.
+#[bedrock_error]
```

Comment:
nit. @lukejmann I don't think this needs to be a `bedrock_error`. this is for internal use only right?